libs/sis-service/src/main/java/uk/gov/di/ipv/core/library/sis/service/SisService.java

```diff
@@ -1,29 +1,341 @@
+import uk.gov.di.model.ContraIndicator;
+
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static software.amazon.awssdk.utils.CollectionUtils.isNullOrEmpty;
+import static uk.gov.di.ipv.core.library.domain.Cri.CIMIT;
+import static uk.gov.di.ipv.core.library.evcs.enums.EvcsVCState.CURRENT;
+import static uk.gov.di.ipv.core.library.evcs.enums.EvcsVCState.PENDING_RETURN;
 
 public class SisService {
     private final SisClient sisClient;
+    private static final Logger LOGGER = LogManager.getLogger();
+    private final ConfigService configService;
+    private final AuditService auditService;
+    private final CimitUtilityService cimitUtilityService;
+    private final UserIdentityService userIdentityService;
+    private final VotMatcher votMatcher;
+    private final EvcsService evcsService;
+    private final CriResponseService criResponseService;
 
+    @SuppressWarnings("java:S107") // Methods should not have too many parameters
     @ExcludeFromGeneratedCoverageReport
-    public SisService(SisClient sisClient) {
+    public SisService(
+            SisClient sisClient,
+            ConfigService configService,
+            AuditService auditService,
+            CimitUtilityService cimitUtilityService,
+            UserIdentityService userIdentityService,
+            VotMatcher votMatcher,
+            EvcsService evcsService,
+            CriResponseService criResponseService) {
         this.sisClient = sisClient;
+        this.configService = configService;
+        this.auditService = auditService;
+        this.cimitUtilityService = cimitUtilityService;
+        this.userIdentityService = userIdentityService;
+        this.votMatcher = votMatcher;
+        this.evcsService = evcsService;
+        this.criResponseService = criResponseService;
     }
 
     @ExcludeFromGeneratedCoverageReport
     public SisService(ConfigService configService) {
+        this.configService = configService;
         this.sisClient = new SisClient(configService);
+        this.userIdentityService = new UserIdentityService(configService);
+        this.auditService = AuditService.create(configService);
+        this.criResponseService = new CriResponseService(configService);
+        this.cimitUtilityService = new CimitUtilityService(configService);
+        this.evcsService = new EvcsService(configService);
+        this.votMatcher =
+                new VotMatcher(
+                        userIdentityService, new Gpg45ProfileEvaluator(), cimitUtilityService);
     }
 
     public SisGetStoredIdentityResult getStoredIdentity(
             ClientOAuthSessionItem clientOAuthSessionItem) {
+        // We want to get any identity in SIS so we ask for all vots.
         return sisClient.getStoredIdentity(
                 clientOAuthSessionItem.getEvcsAccessToken(),
-                clientOAuthSessionItem.getVtrAsVots(),
+                List.of(Vot.P0, Vot.P1, Vot.P2, Vot.P3),
```

Comment:
I might be missing something but I thought we would pass the VTR as is? since we want to verify SIS's calculated VOT is the same as our calculated VOT for this reuse of identity